### PR TITLE
Project tidy

### PR DESCRIPTION
**Pull request by MFAshby**
Hi!

Fennel looks like an awesome project, I'm about to try it out for self-hosting my own calendar and contacts on some low-power  hardware I have :) 

I struggled initially to build the project, there wasn't a simple script and the Makefile was empty. Looking deeper, it seems the project started pre `go mod` becoming popular. 

I've rearranged the project files slightly according to [some unofficial guidelines](https://github.com/golang-standards/project-layout), updated the project to use `go.mod` instead of vendoring dependencies, and added some commands to the Makefile &  README to help others get started. This also makes the travis CI config a lot simpler.

### CODE REVIEW

**Comment by MFAshby** [comment]
This would probably fix https://github.com/swordlordcodingcrew/fennel/issues/4

**Comment by LordEidi** [comment]
Am currently looking through your changes. While doing so I noticed, that there is still a Travis build file. That could be removed. If I automatically build in the future, I would do so on Github anyway.

**Review by LordEidi — COMMENTED** [review]
Let's remove that file altogether
.

File: fennelcore/db.go [hunk 1]
```diff
@@ -71,25 +80,12 @@ func InitDatabase(){
...
-		scope.SetColumn("UpdDat", time.Now())
-	}
-	*/
-}
-
 //
 func CloseDB() {
-
-	db.Close()
+	// db.Close()
```
Comment (LordEidi) [inline fennelcore/db.go:85]: Why no DB close?

File: fennelcore/db.go [hunk 2]
```diff
@@ -30,37 +31,45 @@ package fennelcore
...
+		dialector = postgres.Open(args)
+	case "mysql":
+		dialector = mysql.Open(args)
+	default:
+		log.Fatalf("Unsupported database dialect %v", dialect)
+	}
+	
+	database, err := gorm.Open(dialector, &gorm.Config{
+		NamingStrategy: schema.NamingStrategy{
+			TablePrefix: "fen_", // Avoid naming collisions with reserved tables like 'user'
```
Comment (LordEidi) [inline fennelcore/db.go]: Changed to fnl

File: go.sum [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+cloud.google.com/go/pubsub v1.0.1/go.mod h1:R0Gpsv3s54REJCy4fxDixWD93lHJMoZTyQ2kNxGRt3I=
+cloud.google.com/go/pubsub v1.1.0/go.mod h1:EwwdRX2sKPjnvnqCa270oGRyludottCI76h+R3AArQw=
+cloud.google.com/go/pubsub v1.2.0/go.mod h1:jhfEVHT8odbXTkndysNHCcx0awwzvfOlguIAii9o8iA=
+cloud.google.com/go/pubsub v1.3.1/go.mod h1:i+ucay31+CNRpDW4Lu78I4xXG+O1r/MAHgjpRVR+TSU=
+cloud.google.com/go/storage v1.0.0/go.mod h1:IhtSnM/ZTZV8YYJWCY8RULGVqBDmpoyjwiyrjsg+URw=
+cloud.google.com/go/storage v1.5.0/go.mod h1:tpKbwo567HUNpVclU5sGELwQWBDZ8gh0ZeosJ0Rtdos=
+cloud.google.com/go/storage v1.6.0/go.mod h1:N7U0C8pVQ/+NIKOBQyamJIeKQKkZ+mxpohlUTyfDhBk=
+cloud.google.com/go/storage v1.8.0/go.mod h1:Wv1Oy7z6Yz3DshWRJFhqM/UCfaWIRTdp0RXyy7KQOVs=
+cloud.google.com/go/storage v1.10.0/go.mod h1:FLPqc6j+Ki4BU591ie1oL6qBQGu2Bl/tZ9ullr3+Kg0=
+dmitri.shuralyov.com/gpu/mtl v0.0.0-20190408044501-666a987793e9/go.mod h1:H6x//7gZCb22OMCxBHrMx7a5I7Hp++hsVxbQ4BYO7hU=
```
Comment (LordEidi) [inline go.sum:40]: Check where that is coming from

File: go.sum [hunk 4]
```diff
@@ -0,0 +1,817 @@
...
+github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
+github.com/BurntSushi/xgb v0.0.0-20160522181843-27f122750802/go.mod h1:IVnqGOEym/WlBOVXweHU+Q+/VP0lqqI8lqeDx9IjBqo=
+github.com/Jeffail/gabs v1.4.0 h1://5fYRRTq1edjfIrQGvdkcd22pkYUrHZ5YC/H2GJVAo=
+github.com/Jeffail/gabs v1.4.0/go.mod h1:6xMvQMK4k33lb7GUUpaAPh6nKMmemQeg5d4gn7/bOXc=
+github.com/Masterminds/semver/v3 v3.1.1/go.mod h1:VPu/7SZ7ePZ3QOrcuXROw5FAcLl4a0cBrbBpGY/8hQs=
+github.com/PuerkitoBio/goquery v1.5.1/go.mod h1:GsLWisAFVj4WgDibEWF4pvYnkVQBpKBKeU+7zCJoLcc=
+github.com/Shopify/sarama v1.19.0/go.mod h1:FVkBWblsNy7DGZRfXLU0O9RCGt5g3g3yEuWXgklEdEo=
```
Comment (LordEidi) [inline go.sum:47]: Kafka?

**Comment by MFAshby** [comment]
Thanks very much for the review & merge @LordEidi !

I'll raise follow-ups for some of the comments you've left if I can.

**Pull request merged**